proposals/lip-0047.md

```diff
@@ -511,21 +647,29 @@ chainProperties.roundEndHeight = g.header.height + len(snapshotStore(0).addresse
 BFTThreshold = snapshotStore(0).threshold
 setBFTParameters(BFTThreshold, BFTThreshold, snapshotStore(0).validators)
 
+# Pass the BLS keys and generator keys to the Validators module
+for every validator element in genesisBlockAssetObject.validatorSubstore.validators:
+    registerValidatorKeys(validator.address, validator.proofOfPossession, validator.generatorKey, validator.blsKey)
+
 # Pass the list of validators to the Validators module
 for i in range(len(snapshotStore(0).validators)):
     addresses = snapshotStore(0).address[i]
 
 setGeneratorList(addresses)
 ```
-
 where:
 
+* `registerValidatorKeys` is defined in the [Validators module][research:validators-module#register-validator-keys].
 * `setBFTParameters` is a function exposed by the [BFT module][research:bft_module].
 * `setGeneratorList`is a function exposed by the [Validators module][research:validators-module].
 
-#### After Block Execution
+### Block Processing
+
+The following steps are executed as part of the block processing, see the [LIP 0055][lip-0055] for details.
+
+#### After Transactions Execution
 
-After a block `b` is executed, the following logic is executed:
+After the transactions in a block `b` are executed, the following logic is executed:
 
 ```python
```

Comment:
I think also after the pseudocode below we should list all functions that are used from other modules there. Currently, `setGeneratorList` does not appear in the list. In general, I would find it easier to once define the notation `moduleName.fct` and mention that in this LIP you call functions from the BFT, Random and Validator module. Then it the notation like `validatorsModule.setGeneratorList` or `Validators.setGeneratorList` makes it directly clear that this is not a local function defined in this LIP but an API.

Reply:
Can be resolved.